kong/utils/public.lua

```diff
@@ -0,0 +1,25 @@
+local pcall = pcall
+local ngx_log = ngx.log
+local ERR = ngx.ERR
+
+
+local _M = {}
+
+
+do
+  local ngx_req_get_post_args = ngx.req.get_post_args
+
+  function _M.get_post_args()
+    local ok, res, err = pcall(ngx_req_get_post_args)
+
+    if not ok or err then
+      ngx_log(ERR, "could not get body args: ", err)
```

Comment:
I think we are missing the case where the error is in `res`?

I think we should either your original approach, either if you were worried about my comment (which was mentioned as low priority), then what I meant was:

```lua
if not ok then
  ngx_log(ERR, "could not get body args: ", res)
  return {}
end

if not res then
  ngx_log(ERR, "could not get body args: ", err)
  return {}
end

return res
```